tanjun/dependencies/limiters.py

```diff
@@ -111,6 +116,39 @@ async def increment_cooldown(self, bucket_id: str, ctx: tanjun_abc.Context, /) -
         """
 
 
+class AbstractConcurrencyLimiter(abc.ABC):
+    """Interface used for limiting command concurrent usage."""
+
+    __slots__ = ()
+
+    @abc.abstractmethod
+    async def try_acquire(self, bucket_id: str, ctx: tanjun_abc.Context, /) -> bool:
+        """Try to acquire a concurrency lock on a bucket.
+
+        Parameters
+        ----------
+        bucket_id : str
+            The concurrency bucket to acquire.
+        ctx : tanjun.abc.Context
+            The context to acquire this resource lock with.
+
+        Returns
+        -------
+        bool
+            Whether the lock was acquired.
+        """
+
+    @abc.abstractmethod
+    async def release(self, bucket_id: str, ctx: tanjun_abc.Context, /) -> None:
+        """Release a concurrency lock on a bucket.
+
+        Raises
+        ------
+        RuntimeError
+            If nothing has this resource acquired.
+        """
```

Comment:
Probably don't want to document this since its standard impl detail which indicates a bug